### PR TITLE
Reduce flakiness in ClickHouse tests

### PR DESCRIPTION
**Pull request by rafiramadhana**
## What does this PR do?

Reduce flakiness in ClickHouse tests by replacing `performCRUD` with `performCRUDWithRetry`.

## Related issues

- Closes https://github.com/testcontainers/testcontainers-go/issues/1898

### CODE REVIEW

File: modules/clickhouse/clickhouse_test.go
Comment (mdelapenya) [inline modules/clickhouse/clickhouse_test.go:220]: Could we avoid named return parameters? There is no specific contributing rule for it, although I do prefer not using them if possible (we do have it in a few places, planned to refactor it at some point)

Comment (mdelapenya) [reply modules/clickhouse/clickhouse_test.go:220]: I'd also merge both methods into just one, as we are not going to use `performCRUD` by itself.

**Review by mdelapenya — COMMENTED** [review]
This LGTM, although I'd like to know your thoughts on the comments I left.

Pending my final approval until then.

Thank you very much for taking the time to contribute this!

File: modules/clickhouse/clickhouse_test.go
Comment (rafiramadhana) [reply modules/clickhouse/clickhouse_test.go:220]: > Could we avoid named return parameters?

ok, i will update the method

> I'd also merge both methods into just one, as we are not going to use performCRUD by itself.

i see. i was avoiding to delete methods because im new to this code base

but, if you say so, ok i'll merge them

Comment (mdelapenya) [reply modules/clickhouse/clickhouse_test.go]: No worries, as the methods are used only within the tests, so nobody will be affected by the change.

File: modules/clickhouse/clickhouse_test.go
Comment (mdelapenya) [inline modules/clickhouse/clickhouse_test.go:228]: Given we already use backoff package, let's use it here too. Not a blocker, though

Comment (rafiramadhana) [reply modules/clickhouse/clickhouse_test.go:228]: ok, on it

Comment (rafiramadhana) [reply modules/clickhouse/clickhouse_test.go]: updated @mdelapenya

Comment (mdelapenya) [reply modules/clickhouse/clickhouse_test.go]: Thanks! The CI is running. Will approve it after the CI passes, thanks!

**Comment by mdelapenya** [comment]
Could you run `make lint` in the clickhouse directory? It seems there is something wrong with the format. 🙏

**Comment by rafiramadhana** [comment]
> Could you run `make lint` in the clickhouse directory? It seems there is something wrong with the format. 🙏

updated https://github.com/testcontainers/testcontainers-go/pull/1902/commits/a941bb212ddb46fe62aa127a1a46f6338c6df284

**Review by mdelapenya — APPROVED** [review]
LGTM, thanks for making the test suite more reliable!!!

**Pull request merged**
